src/caret_analyze/plot/visualize_lib/bokeh/message_flow.py

```diff
@@ -122,124 +120,160 @@ def create_figure(self) -> Figure:
                 source=source,
                 x_range_name=x_range_name
             )
+            fig.add_tools(flow_source.create_hover({'renderers': [line]}))
 
         return fig
 
 
-class Offset:
+class MessageFlowRectSource:
+    """Class to generate message flow rect sources."""
+
     def __init__(
         self,
-        offset_ns: int
+        target_path: Path
     ) -> None:
-        self._offset = offset_ns
+        self._target_path = target_path
+        self._hover_keys = HoverKeysFactory.create_instance('message_flow_rect', target_path)
+        self._hover_source = HoverSource(self._hover_keys)
 
-    def __str__(self) -> str:
-        return self._str
+    def create_hover(self, options: Dict[str, Any] = {}) -> HoverTool:
```

Comment:
The purpose of this is to have support for all of the following options.
https://docs.bokeh.org/en/latest/docs/reference/models/tools.html#

renderers | An explicit list of renderers to hit test against.
-- | --
callback | A callback to run in the browser whenever the input's value changes.
tooltips | The (name, field) pairs describing what the hover tool should display when there is a hit.
formatters | Specify the formatting scheme for data source columns, e.g.
mode | Whether to consider hover pointer as a point (x/y values), or a span on h or v directions.
muted_policy | Whether to avoid showing tooltips on muted glyphs.
point_policy | Whether the tooltip position should snap to the "center" (or other anchor) position of the associated glyph, or always follow the current mouse cursor position.
line_policy | Specifies where the tooltip will be positioned when hovering over line glyphs:
anchor | If point policy is set to "snap_to_data", anchor defines the attachment point of a tooltip.
attachment | Whether the tooltip should be displayed to the left or right of the cursor position or above or below it, or if it should be automatically placed in the horizontal or vertical dimension.
show_arrow | Whether tooltip's arrow should be shown.

[renderers](https://docs.bokeh.org/en/latest/docs/reference/models/tools.html#bokeh.models.HoverTool.renderers)

An explicit list of renderers to hit test against.

[callback](https://docs.bokeh.org/en/latest/docs/reference/models/tools.html#bokeh.models.HoverTool.callback)

A callback to run in the browser whenever the input's value changes.

[tooltips](https://docs.bokeh.org/en/latest/docs/reference/models/tools.html#bokeh.models.HoverTool.tooltips)

The (name, field) pairs describing what the hover tool should display when there is a hit.

[formatters](https://docs.bokeh.org/en/latest/docs/reference/models/tools.html#bokeh.models.HoverTool.formatters)

Specify the formatting scheme for data source columns, e.g.

[mode](https://docs.bokeh.org/en/latest/docs/reference/models/tools.html#bokeh.models.HoverTool.mode)

Whether to consider hover pointer as a point (x/y values), or a span on h or v directions.

[muted_policy](https://docs.bokeh.org/en/latest/docs/reference/models/tools.html#bokeh.models.HoverTool.muted_policy)

Whether to avoid showing tooltips on muted glyphs.

[point_policy](https://docs.bokeh.org/en/latest/docs/reference/models/tools.html#bokeh.models.HoverTool.point_policy)

Whether the tooltip position should snap to the "center" (or other anchor) position of the associated glyph, or always follow the current mouse cursor position.

[line_policy](https://docs.bokeh.org/en/latest/docs/reference/models/tools.html#bokeh.models.HoverTool.line_policy)

Specifies where the tooltip will be positioned when hovering over line glyphs:

[anchor](https://docs.bokeh.org/en/latest/docs/reference/models/tools.html#bokeh.models.HoverTool.anchor)

If point policy is set to "snap_to_data", anchor defines the [attachment](https://docs.bokeh.org/en/latest/docs/reference/models/tools.html#bokeh.models.HoverTool.attachment) point of a tooltip.

attachment

Whether the tooltip should be displayed to the left or right of the cursor position or above or below it, or if it should be automatically placed in the horizontal or vertical dimension.

[show_arrow](https://docs.bokeh.org/en/latest/docs/reference/models/tools.html#bokeh.models.HoverTool.show_arrow)

Whether tooltip's arrow should be shown.

Reply:
Noted.
If it is compliant with the bokeh documentation, I consider that it is not a problem.